gcc/rust/backend/rust-compile-expr.h

```diff
@@ -45,7 +47,11 @@ class CompileExpr : public HIRCompileBase
     HIR::Expr *tuple_expr = expr.get_tuple_expr ().get ();
     TupleIndex index = expr.get_tuple_index ();
 
-    Bexpression *receiver_ref = CompileExpr::Compile (tuple_expr, ctx);
+    Bexpression *receiver_ref
+      = CompileExpr::Compile (tuple_expr, ctx, &terminated);
```

Comment:
Please remove the unreachable code termination in GENERIC generation, GCC should be able to optimize this. It is worth while doing but it's not the right time to be adding this to the GENERIC backend yet.

Reply:
The reason why I add it is that I'm afraid that the following code will mess up the GENERIC tree.
```rust
fn main() {
    let a: (i32, i32, i32) = if true {
        return;
        // Here, we'll assign `()` to `(i32, i32, i32)`. 
    } else {
        (1, 2, 3)
    };
}
```
Are you sure there is no problems at all? If there are no problems, I'll remove the unreachable code termination in GENERIC generation. (Maybe I'll have a try later.)

Reply:
It shouldn't cause issues you can do it in C/C++ FE and they directly create GENERIC tress. Even if it did cause issues it shouldn't be in this PR since this is an optimization outside of the scope of what this PR is about and it is something we should discuss/design first on zulip or a github iussue.

Reply:
Yeah, it is OK now since only fairly limited part of NeverType is introduced. There is a different case when I was writing this code.

Reply:
Yeah, it is OK now since only fairly limited part of NeverType is introduced. There is a different case when I was writing this code.